Assets/MixedRealityToolkit/Inspectors/Profiles/BaseMixedRealityProfileInspector.cs

```diff
@@ -55,18 +56,41 @@ protected static bool RenderProfile(SerializedProperty property, GUIContent guiC
         /// <param name="property">the <see cref="BaseMixedRealityProfile"/> property.</param>
         /// <param name="showAddButton">Optional flag to hide the create button.</param>
         /// <returns>True, if the profile changed.</returns>
-        protected static bool RenderProfile(SerializedProperty property, bool showAddButton = true)
+        protected static bool RenderProfile(SerializedProperty property, bool showAddButton = true, Type serviceType = null)
         {
-            return RenderProfileInternal(property, null, showAddButton);
+            return RenderProfileInternal(property, null, showAddButton, serviceType);
         }
 
-        private static bool RenderProfileInternal(SerializedProperty property, GUIContent guiContent, bool showAddButton)
+        private static bool RenderProfileInternal(SerializedProperty property, GUIContent guiContent, bool showAddButton, Type serviceType = null)
         {
             bool changed = false;
-            EditorGUILayout.BeginHorizontal();
 
             var oldObject = property.objectReferenceValue;
 
+            // If we're constraining this to a service type, check whether the profile is valid
+            // If it isn't, issue a warning.
+            if (serviceType != null && oldObject != null)
+            {
+                bool profileTypeIsValid = false;
+
+                foreach (MixedRealityServiceProfileAttribute serviceProfileAttribute in oldObject.GetType().GetCustomAttributes(typeof(MixedRealityServiceProfileAttribute), true))
+                {
+                    if (serviceProfileAttribute.ServiceType.IsAssignableFrom(serviceType))
+                    {
+                        profileTypeIsValid = true;
+                        break;
+                    }
+                }
+
+                if (!profileTypeIsValid)
+                {
+                    EditorGUILayout.HelpBox("The profile is not configured to be used with a service of type " + serviceType.Name
+                        + "\nIs your profile missing the " + typeof(MixedRealityServiceProfileAttribute).Name + " attribute?", MessageType.Warning);
```

Comment:
Most consumers of MRTK will not be creating their own profile classes. Suggest the message read something like:

"This profile is not supported for <service>. Using an unsupported service may result in unexpected behavior."